### ISSUE TITLE
PascalCase tags flagged as malformed headings

### ISSUE THREAD

**Issue by mactep** [rvben/rumdl#905]
### rumdl version

0.2.75

### Installation method

Built from source

### Bug description

When MD018's `tags` option is enabled, Obsidian tags whose first character is uppercase `#PascalCase` are still treated as level-1 headings by MD022 and MD025.

### Steps to reproduce

1. test.md

```markdown
# Title

Content

#camelCase
#PascalCase
#snake_case
#kebab-case
```

2. ~/.config/rumdl/rumdl.toml:

```
[MD018]
tags = true
```

3. `rumdl check test.md`

```
test.md:6:1: [MD022] Expected 1 blank line above heading [*]
test.md:6:1: [MD022] Expected 1 blank line below heading [*]
test.md:6:2: [MD025] Multiple top-level headings (level 1) in the same document [*]

Issues: Found 3 issues in 1 file (8ms)
Run `rumdl fmt` to automatically fix all 3 issues
```

### Expected behavior

All four tags should not be flagged.

### Actual behavior

The `#PascalCase` tag is flagged, while the others are clean.

### Configuration

```toml
[MD018]
tags = true
```

### Operating system

Linux

### Additional context

Root cause appears to be the heading-validity heuristic in
`src/lint_context/heading_detection.rs` (~line 773):

```rust
    let is_valid = !spaces_after.is_empty()
        || rest.is_empty()
        || level > 1
        || rest.trim().chars().next().is_some_and(char::is_uppercase);
```

For a single-`#` line with no space, the line only counts as a heading when the first content character is uppercase; lowercase-initial lines are marked `is_valid = false` (paragraph text), which is why MD022/MD025 skip them. MD018 uses the shared, case-agnostic `TAG_PATTERN` from `src/utils/obsidian_tag.rs`, so the two subsystems disagree for uppercase-initial tags.

Reference: [Obsidian docs](https://obsidian.md/help/tags#Tag%20format)
